src/state/usePasscodeAuth/usePasscodeAuth.ts

```diff
@@ -1,9 +1,22 @@
 import { useCallback, useEffect, useState } from 'react';
 import { useHistory } from 'react-router-dom';
+const queryString = require('query-string');
+
+export function getRoomName() {
+  const parsed = queryString.parse(window.location.search);
+  const room = parsed.room ? parsed.room : window.sessionStorage.getItem('room');
+  return room;
+}
+
+export function getUserName() {
+  const parsed = queryString.parse(window.location.search);
+  const user = parsed.user ? parsed.user : window.sessionStorage.getItem('user');
+  return user;
+}
```

Comment:
If we are sharing this url, I would expect we want the user to enter their own username. Why are we setting it in the url? Can you please provide the use case for this?

Reply:
@charliesantos there are a few use-cases where prefilling a user's name could be useful.

For example, I click a button to send an SMS link to the video room. The button trigger's a controller that queries by DB based on the phone entered and pulls the first name based on the phone number. Then, the controller adds that param to the link and sends an SMS -> user has a one-click join experience.

Reply:
That's a good use case @cweems. In that case, I think we can keep this feature in this PR.